### ISSUE TITLE
Arrays of one element are not reduced to a single item

### ISSUE THREAD

**Issue by jason-fox** [Fiware/context.Orion-LD#310]
```console
curl -X POST   'http://localhost:1026/ngsi-ld/v1/entities/'   -H 'Content-Type: application/ld+json'   -d '{
    "id": "urn:ngsi-ld:Building:1234",
    "type": "Building",
     "category": {"type": "Property", "value": ["commercial"]},
    "@context": "https://fiware.github.io/tutorials.Step-by-Step/data-models-context.jsonld"
}'
```

```console
curl http://localhost:1026/ngsi-ld/v1/entities/urn:ngsi-ld:Building:1234
``` 
returns
```json
category":{"type":"Property","value":["commercial"]}
```

should return

```json
category":{"type":"Property","value":"commercial"}
```

From the JSON LD spec:

>  Values of terms associated with an @set or @list container are always represented in the form of an array, even if there is just a single value that would otherwise be optimized to a non-array form in compact form (see § 4.26 Compacted Document Form). This makes post-processing of JSON-LD documents easier as the data is always in array form, even if the array only contains a single value.

Since the data model doesn't specify @set the array should be compacted to a single string in this case

**Comment by kzangeli** [comment]
In the context, if a key is an object and has a member @type, with either the value "@set" or "@list", then the array should remain intact. If not, the array should be simplified if only one item in it.

This should be performed at creation time, before storing the value in the database.

**Comment by jason-fox** [comment]
For clarity, I think you need to support this:

```json
{
  "@context": {
    "xyz": "https:///schema/xyz",
    "alwaysAnArray": {
      "@id": "xyz:array",   
      "@type": "@set"
    }
}
```

So this Property is an array of one

```json
{
    "alwaysAnArray":{
         "type":"Property"," value":["commercial"]
    }
}
```

**Comment by MohamedSadiq102** [comment]
fixed close #310

**Comment by kzangeli** [comment]
This issue is wrong, I'm afraid.
The @context doesn't work on "another level above" - it's always "value" that would need the "@type": "@set" for it to work this way.
I need to undo the modification
